libs/sdk-ui-dashboard/src/model/commandHandlers/alerts/createAlertHandler.ts

```diff
@@ -0,0 +1,37 @@
+// (C) 2021 GoodData Corporation
+import { call, put } from "redux-saga/effects";
+import { dispatchDashboardEvent } from "../../eventEmitter/eventDispatcher";
+import { DashboardContext } from "../../types/commonTypes";
+import { IWidgetAlert, IWidgetAlertDefinition } from "@gooddata/sdk-backend-spi";
+import { CreateAlert } from "../../commands/alerts";
+import { alertCreateError, alertCreateSuccess } from "../../events/alerts";
+import { PromiseFnReturnType } from "../../types/sagas";
+import { alertsActions } from "../../state/alerts";
+
+function createAlert(
+    ctx: DashboardContext,
+    alert: IWidgetAlert | IWidgetAlertDefinition,
+): Promise<IWidgetAlert> {
+    const { backend, workspace } = ctx;
+
+    return backend.workspace(workspace).dashboards().createWidgetAlert(alert);
+}
+
+export function* createDashboardAlertCommandHandler(ctx: DashboardContext, cmd: CreateAlert) {
+    // eslint-disable-next-line no-console
+    console.debug("handling create alert", cmd, "in context", ctx);
+
+    try {
+        const alert: PromiseFnReturnType<typeof createAlert> = yield call(
+            createAlert,
+            ctx,
+            cmd.payload.alert,
+        );
+
+        yield put(alertsActions.addAlert(alert));
+
+        yield dispatchDashboardEvent(alertCreateSuccess(ctx, alert, cmd.correlationId));
+    } catch (e) {
+        yield dispatchDashboardEvent(alertCreateError(ctx, cmd.payload.alert, e, cmd.correlationId));
```

Comment:
don't need custom event for this. there is command failed event already. the INTERNAL_ERROR reason is the right reason